### PR TITLE
Fix some insights query editor bugs and add more tests

### PR DESCRIPTION
**Pull request by rli**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt [hunk 1]
```diff
@@ -143,6 +143,14 @@ class QueryEditor internal constructor(
         return unit.second
     }
 
+    fun setSelectedTimeUnit(unit: ChronoUnit) {
+        for (knownUnits in timeUnits) {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt:147]: nit: seeing a `for` loops is weird, any reason to do this over something more functional like `timeunits.filter { unit == it.second }.forEach{relativeTimeUnit.selectedItem = knownUnits}`?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt [hunk 2]
```diff
@@ -143,6 +143,14 @@ class QueryEditor internal constructor(
         return unit.second
     }
 
+    fun setSelectedTimeUnit(unit: ChronoUnit) {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt:146]: Also uh what does this do?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt:147]: Can we just use an enum instead? And an EnumComboBoxModel?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt (hunk 2 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt:143]: the item being set needs to be equal to something else that's currently in the model, which is impossible if the model holds a bunch of pairs and we're trying to select a ChronoUnit

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/TimeUnit.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import software.aws.toolkits.resources.message
+import java.time.temporal.ChronoUnit
+
+enum class TimeUnit(val unit: ChronoUnit, val text: String) {
+    MINUTES(ChronoUnit.MINUTES, message("cloudwatch.logs.time_minutes")),
+
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/TimeUnit.kt:11]: nit: the spacing is a bit weird, why are there line returns between each entry?

**Review by hunterwerlla — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/TimeUnit.kt (hunk 3 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/TimeUnit.kt]: fixed

**Pull request merged**